UI/importers/classic.cpp

```diff
@@ -0,0 +1,577 @@
+/******************************************************************************
+    Copyright (C) 2019 by Dillon Pentz <dillon@vodbox.io>
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include "importers.hpp"
+
+#include <QByteArray>
+
+using namespace std;
+using namespace json11;
+
+static bool source_name_exists(string name, Json::array sources)
```

Comment:
Should be something like:
```cpp
static bool source_name_exists(const string &name, const Json::array &sources)
```
if you're not modifying that data. If you don't use a reference, those objects will be copied. If you're not modifying the data, then they should also be const.

Reply:
Also there's something to be said about the order of function parameters.  Generally, you want to have consistency in everything you do, including things like the order of function parameters.

In this case, you're checking an array of sources to see if the specified name exists.  Normally, I personally would usually have `sources` as the first parameter, then `name` as the second parameter to sort of communicate that you're checking `sources` for `name`.  Parameters should generally be ordered in relative terms like that.  Most relevance to least relevance, if that makes sense.